src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php

```diff
@@ -53,12 +52,12 @@ public function createArgumentMetadata($controller): array
                     throw new InvalidMetadataException(sprintf('Controller "%s" has more than one attribute for "$%s" argument.', $representative, $param->getName()));
                 }
 
-                if (isset($reflectionAttributes[0])) {
-                    $attribute = $reflectionAttributes[0]->newInstance();
+                foreach ($reflectionAttributes as $reflectionAttribute) {
+                    $attributes[] = $reflectionAttribute->newInstance();
```

Comment:
This means that one cannot annotate their controllers with attributes defined in not-installed (optional) deps.
This would be a problem in libs, but for controllers, I think it's OK.

Reply:
I disagree. We should find a way to only create instances of attributes that are relevant to us. The attribute could be a dev dependency (take the JetBrains attributes for instance) that I don't want to deploy to prod.

Reply:
Added  `hasAttribute(string $type): bool` (which is the only one used in core for now) and made `getAttributes()` returns a list of `\ReflectionAttribute` instances.